custom/icds/data_management/base.py

```diff
@@ -0,0 +1,53 @@
+from corehq.form_processor.backends.sql.dbaccessors import CaseReindexAccessor
+from corehq.util.doc_processor.interface import BulkDocProcessor
+from corehq.util.doc_processor.sql import SqlDocumentProvider
+
+
+class DataManagement(object):
+    slug = ""
+    name = ""
+    case_type = None
+    doc_processor = None
+
+    def __init__(self, domain, *args, **kwargs):
+        self.domain = domain
+
+    def case_accessor(self):
+        raise NotImplementedError
+
+    def run(self, iteration_key):
+        """
+        iterate sql records and update them as and when needed
+        """
+        record_provider = SqlDocumentProvider(iteration_key, self.case_accessor())
+        processor = BulkDocProcessor(record_provider, self.doc_processor(self.domain))
+        processor.run()
+
+
+class SQLBasedDataManagement(DataManagement):
+    def __init__(self, domain, db_alias, from_date=None, till_date=None):
+        super().__init__(domain)
+        self.db_aliases = [db_alias] if db_alias else None
+        self.from_date = from_date
+        self.till_date = till_date
```

Comment:
nit: common convention for date naming is `start_date`, `end_date`

Reply:
https://github.com/dimagi/commcare-hq/pull/26610/commits/d4214e80d4f871bd032ac00fe93345f35bd092ac